linera-views/src/batch.rs

```diff
@@ -158,29 +159,39 @@ impl Batch {
         for op in self.operations {
             match op {
                 WriteOperation::Delete { key } => {
+                    // We delete a key. However if said key was already matched by a delete_prefix then
+                    // nothing needs to be done.
                     if !is_prefix_matched(&delete_prefix_set, &key) {
```

Comment:
Ah, sorry, I meant:
* First `DeletePrefix { key_prefix: "FOO" }`. So `"FOO"` is now in `delete_prefix_set`.
* Next `Put { key: "FOOBAR", value: "BAZ" }`. Now we have an entry `"FOOBAR"` in `delete_and_insert_map`.
* Finally `Delete { key: "FOOBAR" }`. This will now be ignored because `"FOOBAR"` is matched by an entry in `delete_prefix_set`.

So if the prefix deletions are done first, wouldn't `"FOOBAR": "BAZ"` still get inserted, even though it shouldn't, because it should have been deleted?

Reply:
You are right. This is another bug that you identified.

Reply:
And I added a corresponding unit test.